packages/template-chakra-storefront/src/hooks/use-manual-bonus-products.js

```diff
@@ -0,0 +1,395 @@
+     */
+    const createManualBonusProductCollections = useCallback(
+        (qualifyingProductToBonusProducts) => {
+            if (
+                !qualifyingProductToBonusProducts ||
+                Object.keys(qualifyingProductToBonusProducts).length === 0
+            ) {
+                return
+            }
+
+            Object.entries(qualifyingProductToBonusProducts).forEach(
+                ([regularProductId, bonusProductsOrTrimOperation]) => {
+                    if (bonusProductsOrTrimOperation?.action === 'trim') {
+                        // Handle trimming operation
+                        const {quantityReduction, newQuantity} = bonusProductsOrTrimOperation
+                        trimManualBonusProductCollection(
+                            regularProductId,
+                            quantityReduction,
+                            newQuantity
+                        )
+                    } else if (
+                        Array.isArray(bonusProductsOrTrimOperation) &&
+                        bonusProductsOrTrimOperation.length > 0
+                    ) {
+                        // Handle adding bonus products
+                        setManualBonusProductCollections((prev) => ({
+                            ...prev,
+                            [regularProductId]: [
+                                ...(prev[regularProductId] || []),
+                                ...bonusProductsOrTrimOperation.map((bonusProduct) => ({
+                                    itemId: bonusProduct.itemId,
+                                    productId: bonusProduct.productId,
+                                    productName: bonusProduct.productName,
+                                    quantity: bonusProduct.quantity,
+                                    addedAt: new Date().toISOString(),
+                                    // Include any price adjustments that show it's a bonus
+                                    priceAdjustments: bonusProduct.priceAdjustments?.filter(
+                                        (adj) => adj.appliedDiscount?.type === 'bonus'
+                                    ),
+                                    // Include promotion information for better tracking
+                                    promotionId: bonusProduct.promotionId,
+                                    // Include bonus discount line item IDs for complete tracking
+                                    bonusDiscountLineItemId: bonusProduct.bonusDiscountLineItemId,
+                                    bonusDiscountPromotionId: bonusProduct.bonusDiscountPromotionId
+                                }))
+                            ]
+                        }))
+                    }
+                }
+            )
+        },
+        [trimManualBonusProductCollection]
+    )
+
+    /**
+     * Creates a manual bonus product collection for a single qualifying product (legacy method)
```

Comment:
What does legacy method mean here?
```suggestion
     * Creates a manual bonus product collection for a single qualifying product (legacy method)
```

Reply:
The comment was generated by Cursor, I have removed the legacy wording.